### ISSUE TITLE
Drop Python2 support

### ISSUE THREAD

**Issue by xLinkOut** [malwaredllc/byob#377]
I'm studying BYOB code to better understanding some cryptography mechanisms for an University course. I was wondering if, knowing that the Python Software Foundation officially discontinued Python 2.x, it was appropriate to remove the support here as well. There are a lot of tricks and tweaks just to ensure backwards compatibility for Python 2 (i.e. `raw_input = input`), and that affect the readability and maintainability of the code itself.

**Comment by malwaredllc** [comment]
Yes, that will need to be done eventually.

Just curious, is BYOB included as a project in your university course, or did do you decide to look into this on your own?

**Comment by xLinkOut** [comment]
Okay! Maybe I could help, I'm getting deeper and deeper into the BYOB code and getting familiar with its various parts. An installation guide would also be very useful, I admit I struggled a bit to run it.

No, unfortunately BYOB is not included as a project in the course. I discovered BYOB long before taking this course but I didn't have the tools to fully understand it. However, I thought of proposing it to the teacher to integrate it, perhaps, in the next year of lessons. It is very educational and allows you to "get your hands dirty" in an attack or defense scenario on the network.
